proto-gen/codec_test.go

```diff
@@ -0,0 +1,55 @@
+package proto_gen
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+func TestCodecMarshallAndUnmarshall_jaeger_type(t *testing.T) {
```

Comment:
are you sure tests are being executed in this proto-gen package? I thought we were excluding them (you have a fmt issue above so pretty sure linter is not running here). I think it's better not to mix generated code with manual code.

Reply:
To what package should we move this then? I can think of `model` only.  

> I think it's better not to mix generated code with manual code.

We already do mix packages e.g. `model` package.

Reply:
in the model we do it out of necessity, since gen-ed code doubles as domain model and the custom types ser/deset methods need to be there. But we usually don't write manual code in ***-gen packages.

You can create a new package like pkg/gogocodec

Reply:
The best would be get rid of this, I cannot find custom codec in OTELcol, so perhaps it's doable to do it without it.

Reply:
The custom codec is needed only if a custom type (e.g. `TaceID) is used directly in the request (probably also response) object - https://github.com/jaegertracing/jaeger-idl/blob/master/proto/api_v2/query.proto#L38 via https://github.com/jaegertracing/jaeger-idl/blob/master/proto/api_v2/query.proto#L128. If the `TraceID` was wrapped into another message the serialization would work without the custom codec.